workflows/seqinspector.nf

```diff
@@ -50,45 +57,162 @@ workflow SEQINSPECTOR {
     ch_methods_description                = Channel.value(
         methodsDescriptionText(ch_multiqc_custom_methods_description))
 
-    ch_multiqc_files = ch_multiqc_files.mix(
+    ch_multiqc_extra_files = ch_multiqc_extra_files.mix(
         ch_workflow_summary.collectFile(name: 'workflow_summary_mqc.yaml'))
-    ch_multiqc_files = ch_multiqc_files.mix(ch_collated_versions)
-    ch_multiqc_files = ch_multiqc_files.mix(
+    ch_multiqc_extra_files = ch_multiqc_extra_files.mix(ch_collated_versions)
+    ch_multiqc_extra_files = ch_multiqc_extra_files.mix(
         ch_methods_description.collectFile(
             name: 'methods_description_mqc.yaml',
             sort: true
         )
     )
 
-    MULTIQC (
-        ch_multiqc_files.collect(),
+    MULTIQC_GLOBAL (
+        ch_multiqc_files
+            .map { meta, file -> file }
+            .mix(ch_multiqc_extra_files)
+            .collect(),
+        ch_multiqc_config.toList(),
+        Channel.empty().toList(),
+        ch_multiqc_logo.toList()
+    )
+
+    // Generate reports by lane
+    multiqc_extra_files_per_lane = ch_multiqc_files
+        .filter { meta, sample -> meta.lane }
+        .map { meta, sample -> [lane: meta.lane] }
+        .unique()
+        .combine(ch_multiqc_extra_files)
+
+    lane_mqc_files = ch_multiqc_files
+        .filter { meta, sample -> meta.lane }
+        .mix(multiqc_extra_files_per_lane)
+        .map { meta, sample -> [ "[LANE:${meta.lane}]", meta, sample ] }
+        .groupTuple()
+        .tap { mqc_by_lane }
+        .collectFile{
+            lane, meta, samples -> [
+                "${lane}_multiqc_extra_config.yml",
+                """
+                    |output_fn_name: \"${lane}_multiqc_report.html\"
+                    |data_dir_name:  \"${lane}_multiqc_data\"
+                    |plots_dir_name: \"${lane}_multiqc_plots\"
+                """.stripMargin()
+            ]
+        }
+        .map { file -> [ (file =~ /(\[LANE:.+\])/)[0][1], file ] }
+        .join(mqc_by_lane)
+        .multiMap { lane, config, meta , samples_per_lane ->
+            samples_per_lane: samples_per_lane
+            config: config
+        }
+
+    MULTIQC_PER_LANE(
+        lane_mqc_files.samples_per_lane,
+        ch_multiqc_config.toList(),
+        lane_mqc_files.config,
+        ch_multiqc_logo.toList()
+    )
+
+    // Generate reports by group
+    multiqc_extra_files_per_group = ch_multiqc_files
+        .filter { meta, sample -> meta.group }
+        .map { meta, sample -> meta.group }
+        .unique()
+        .map { group -> [group:group] }
+        .combine(ch_multiqc_extra_files)
+
+    group_mqc_files = ch_multiqc_files
+        .filter { meta, sample -> meta.group }
+        .mix(multiqc_extra_files_per_group)
+        .map { meta, sample -> [ "[GROUP:${meta.group}]", meta, sample ] }
+        .groupTuple()
+        .tap { mqc_by_group }
+        .collectFile{
```

Comment:
Is it really required to construct a custom MultiQC config just to set the output file paths? I somehow think that it should be possible to handle that in the `publishDir` of the `module.config`? Or am I missing something?

Reply:
The issue here is that if you run the MultiQC module once for each lane without specifying different configs each time, it will create files with the same name regardless the lane number. Since the filename is all you have to play with when setting `publishDir`, it becomes very hard to sort them out into different folders.